solutions/src/main/java/ru/sberbank/school/task08/SerializableManager.java

```diff
@@ -17,28 +18,41 @@ public SerializableManager(@NonNull String filesDirectoryPath) {
 
     @Override
     public void initialize() {
-        throw new UnsupportedOperationException("Implement me!");
     }
 
     @Override
-    public void saveGame(String filename, Savable gameState) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public void saveGame(@NonNull String filename, @NonNull MapState<GameObject> gameState) throws SaveGameException {
+        try (FileOutputStream fos = new FileOutputStream(filesDirectory + File.separator + filename);
+             ObjectOutputStream out = new ObjectOutputStream(fos)) {
+            out.writeObject(gameState);
+        } catch (FileNotFoundException e) {
+            throw new SaveGameException("File not found");
+        } catch (IOException ex) {
+            throw new SaveGameException("Stream reading is failed");
+        }
     }
 
     @Override
-    public Savable loadGame(String filename) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public MapState<GameObject> loadGame(@NonNull String filename) throws SaveGameException {
+        try (FileInputStream fis = new FileInputStream(filesDirectory + File.separator + filename);
+             ObjectInputStream in = new ObjectInputStream(fis)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 13, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-702531695>*